src/main/java/akshan/task/Event.java

```diff
@@ -0,0 +1,49 @@
+package akshan.task;
+
+/**
+ * Represents an event task with a start and end time.
+ */
+public class Event extends Task {
+    protected final String start;
+    protected final String end;
+
+    /**
+     * Constructs an Event task.
+     *
+     * @param name Name of the event.
+     * @param start Date when the event begins.
+     * @param end Date when the event ends.
+     * @throws IllegalArgumentException If name, start, or end is null or empty.
+     */
+    public Event(String name, String start, String end) throws IllegalArgumentException {
+        super(name);
+        if (start == null || start.trim().isEmpty() || end == null || end.trim().isEmpty()) {
```

Comment:
Maybe you can extract the condition out as a boolean variable since it looks to be [quite long](https://nus-cs2113-ay2425s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-complicated-expressions)

Reply:
True, had a hard time reading this myself haha. Will refactor this in the next commit.